### ISSUE TITLE
compile from source fails on raspberry

### ISSUE THREAD

**Issue by cutnicace** [getdnsapi/stubby#143]
Hi I would like to use stubby on my raspberry setup which already runs a PiHole instance (hw is a pi3 with the current Raspbian Stretch Lite Release date:2018-11-13).

All the how_to's I found regarding that topic suggest the following approach:

    sudo apt-get install libtool autoconf m4 libssl-dev libyaml-dev

    git clone https://github.com/getdnsapi/getdns.git
    cd getdns
    git checkout develop
    git submodule update --init
    libtoolize -ci 
    autoreconf -fi
    mkdir -v build && cd build
    ../configure --prefix=/usr/local --without-libidn --without-libidn2 --enable-stub-only --with-ssl --with-stubby
    make
    sudo make install

but regardless which way I follow, when I get to the make part, I get:

    pi@PiHole:~/getdns/build $ make
    cd src && make default
    make[1]: Entering directory '/home/pi/getdns/build/src'
    ../libtool --quiet --tag=CC --mode=compile gcc -I../../src -I. -I../../src/util/auxiliary -I../../src/../stubby/src -g -O2 -Wall -Wextra -D_BSD_SOURCE -D_DEFAULT_SOURCE   -Wpedantic -c ../../src/convert.c -o convert.lo
    In file included from ../../src/convert.c:50:0:
    ../../src/util-internal.h: In function ‘_getdns_tls_version2openssl_version’:
    ../../src/util-internal.h:229:29: error: ‘TLS1_3_VERSION’ undeclared (first use in this function)
      case GETDNS_TLS1_3: return TLS1_3_VERSION;
                                 ^~~~~~~~~~~~~~
    ../../src/util-internal.h:229:29: note: each undeclared identifier is reported only once for each function it appears in
    Makefile:122: recipe for target 'convert.lo' failed
    make[1]: *** [convert.lo] Error 1
    make[1]: Leaving directory '/home/pi/getdns/build/src'
    Makefile:53: recipe for target 'default' failed
    make: *** [default] Error 2

What is wrong here? 
When I inspect the referenced file I can confirm that the line with the "TLS1_3_VERSION" is the only ocurrence in the whole file, but what does the compiler want to tell me with this?

Can I do anything to fix it by myself, or is there a bug in the src files?

**Comment by wtoorop** [comment]
Thanks cutnicace,  May I ask which howto's you were looking at?
The `git checkout develop` statement is a mistake.  That gives you the branch on which development takes place. The latest released getdns and stubby is available with `git checkout master`.

**Comment by cutnicace** [comment]
Hey wtoorop, Thanks for the quick reply. 
I started with a article in a computer magazine "c't" but that was already a few months old, so I went looking in the web and found: 

https://dnsprivacy.org/wiki/pages/viewpage.action?pageId=3145786
and 
https://www.reddit.com/r/pihole/comments/7oyh9m/guide_how_to_use_pihole_with_stubby/

which are also some months old but all of them had similar approaches for the install.
I was also a little hesistant to checkout the develop branch, but since all three sources did it that way I assumed it was the preferred way to go.

Thanks, the compile went through without error's this time!
